spaghetti/util.py

```diff
@@ -720,7 +720,7 @@ def _points_as_gdf(
     # instantiate geopandas.GeoDataFrame
     pts_list = list(pts_dict.items())
     points = geopandas.GeoDataFrame(pts_list, columns=[id_col, geom_col])
-    points.geometry = points.geometry.apply(lambda p: Point(p))
+    points.geometry = points[geom_col].map(lambda p: Point(p))
```

Comment:
This doesn't feel right. What is `pts_dict.items()` exactly? You should not initiate a GeoDataFrame without a geometry column. One way of refactoring this would be 

```py
points = pandas.DataFrame(pts_list, columns=[id_col, geom_col])
points = points.set_geometry(points[geom_col].map(lambda p: Point(p)))
```

Plus you shall also be able to avoid that loop over scalar `Point` constructor either with `points_from_xy` or with `shapely.points`.

Reply:
Awesome! I will give that a try next! Thanks @martinfleis!

Reply:
Also, the `pts_dict` is dictionary where keys are IDs and values are coordinates. This can either be coming from a `PointPattern` object or mimic one.